### PR TITLE
pythonPackages.missingno: init at 0.4.2

### PR DESCRIPTION
**Pull request by melsigl**
###### Motivation for this change
`missingno` is an easy to use Python package for data visualization, especially regarding messy data with missing vlaues.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

File: pkgs/development/python-modules/missingno/default.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+, matplotlib
+, scipy
+, seaborn
+}:
+
+buildPythonPackage rec {
+  pname = "missingno";
+  version = "0.4.2";
+
+  src = fetchPypi {
```
[the remark this answers is not in the document]
Comment (jonringer) [reply pkgs/development/python-modules/missingno/default.nix]: they are important to ensure your package wont be broken by upstream dependencies, and I find that most science related packages like to break often.

[the remark this answers is not in the document]
Comment (melsigl) [reply pkgs/development/python-modules/missingno/default.nix]: I found that integrating tests here initiates a snowball effect, where python packages are used for tests that are not (yet) in nixos. Dunno how integrating the tests of this `missingno` package will justify this. Thus, we can either ignore those tests, or alternatively, do not merge this package.

[the remark this answers is not in the document]
Comment (jonringer) [reply pkgs/development/python-modules/missingno/default.nix]: the pypi package didn't contain tests, I would do:

```
{ lib
, buildPythonPackage
, fetchFromGitHub
, matplotlib
, scipy
, seaborn
, pytest
}:

buildPythonPackage rec {
  pname = "missingno";
  version = "0.4.2";

  src = fetchFromGitHub {
    owner = "ResidentMario";
    repo = pname;
    rev = version;
    sha256 = "0wxhj7qm2fcaiprdjfrx9p8nikc10qhivf7rzwi04r6mrzyi0i9x";
  };

  propagatedBuildInputs = [
    matplotlib
    scipy
    seaborn
  ];

  checkInputs = [ pytest ];
  checkPhase = ''
    pytest tests/util_tests.py -k 'not cutoff'
  '';

  meta = with lib; {
    description = "Missing data visualization module for Python.";
    homepage = https://github.com/ResidentMario/missingno;
    maintainers = with maintainers; [ msigl ];
    license = licenses.mit;
  };
}
```

[the remark this answers is not in the document]
Comment (melsigl) [reply pkgs/development/python-modules/missingno/default.nix]: Ah ok. Naively I did assume that we have to run all tests. According to this packages' [contributing guidelines](https://github.com/ResidentMario/missingno/blob/0.4.2/CONTRIBUTING.md) there are two types of tests. Namely the visualizing tests `tests/viz_tests.py` and utility tests `tests/util_tests.py.` 
For those viz-tests additional data is needed, which is loaded via quilt - a data versioning system for AWS. However, this quilt is not in nixos as some other dependencies of quilt are also not in nixos.
Long story short: sure, we can only do the utility tests as you suggested.

[the remark this answers is not in the document]
Comment (jonringer) [reply pkgs/development/python-modules/missingno/default.nix]: yea, those looked very involved.

We more or less just want to verify that the package is in a valid state. Util tests should give us some code coverage

**Review by doronbehar — APPROVED** [review]
Passes nixpkgs-review build and diff LGTM.

**Comment by doronbehar** [comment]
Too bad we didn't merge it back then, there's merge conflicts now. @melsigl do you still want to get this in?

File: pkgs/development/python-modules/missingno/default.nix [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  propagatedBuildInputs = [
+    matplotlib
+    scipy
+    seaborn
+  ];
+
+  checkInputs = [ pytest ];
+
+  checkPhase = ''
+  pytest tests/util_tests.py -k 'not cutoff'
```
Comment (doronbehar) [inline pkgs/development/python-modules/missingno/default.nix:30]: Indentation + missing hooks, see https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/explicit-phases.md .

```suggestion
    runHook preCheck

    pytest tests/util_tests.py -k 'not cutoff'

    runHook postCheck
```

**Review by doronbehar — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/missingno/default.nix (hunk 2 above)
Comment (SuperSandro2000) [reply pkgs/development/python-modules/missingno/default.nix:30]: Also pytestCheckHook is prefered.

**Comment by melsigl** [comment]
@doronbehar I do not need that particular package anymore. Thus, we can close this pull request
